tree/30_generic_methods/file_from_string_mustache.cf

```diff
@@ -0,0 +1,46 @@
+#####################################################################################
+# Copyright 2014 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name File from a mustache string
+# @description This is a bundle to build a file from a mustache string
+# @agent_version >=3.7
+#
+# @parameter template        String containing a template to be expanded
+# @parameter destination     Destination file
+#
+# @class_prefix file_from_template
+# @class_parameter destination
+# This bundle will define a class file_from_template_${destination}_{kept,repaired,not_ok,ok,reached}
+
+bundle agent file_from_string_mustache(template, destination)
+{
+  vars:
+    "class_prefix"        string => canonify("file_from_template_${destination}");
```

Comment:
Well unless you have a specific reason for it not to be, this should be the bundle name, so yeah, "file_from_string_mustache"